js/os.ts

```diff
@@ -42,6 +36,26 @@ export function isTTY(rid = 0): boolean {
   return res.isTty();
 }
 
+interface IsTTY {
+  stdin: boolean;
+  stdout: boolean;
+  stderr: boolean;
+}
+
+/** Check if running in terminal.
+ *
+ * isTTY().stdin; // stdin
+ * isTTY().stdout; // stdout
+ * isTTY().stderr; // stderr
+ */
+export function isTTY(): IsTTY {
+  return {
+    stdin: checkTTY(0),
+    stdout: checkTTY(1),
+    stderr: checkTTY(2)
```

Comment:
You should do this in a single op (and cache the result)

Reply:
@hayd done!

Reply:
Nearly there!

By cache the result, I mean have:
```ts
let _isTTY: IsTTY;
export function isTTY(): IsTTY {
  if (_isTTY) return _isTTY;
  ... // builder
  _isTTY = {stdin: res.stdin, ... }
  return _isTTY;
}
```

That way the op is only ever done once (every other time it's taken from the variable).

Reply:
@hayd thanks for help!

Reply:
@hayd done